trie/smt.py

```diff
@@ -335,8 +335,7 @@ def exists(self, key: bytes) -> bool:
         validate_length(key, self._key_size)
 
         try:
-            self.get(key)
-            return True
+            return self.get(key) != self._default
```

Comment:
this line is not actually needed because youll get a KeyError now
However - I find it to be more readable in a way..